src/rpc/server/scala/src/main/scala/com/ctc/omega_edit/grpc/Session.scala

```diff
@@ -328,7 +328,7 @@ class Session(
         offset,
         request.length.getOrElse(0),
         session.search(
-          request.pattern.toString,
+          request.pattern.toString("utf-8"),
```

Comment:
Not all data we'll be editing will be character data that will decode as UTF-8.  We need to handle byte (binary) data also.  I recommend against this conversion.  I have a test case in review that does search and replace with byte data that won't decode into UTF-8.

Reply:
Without this conversion search will not work properly. Without doing the conversion it is sending a `ByteString` and the incorrect pattern length is calculated causing a nullptr to be returned.

Reply:
@scholarsmate Do we maybe want to make that a separate issue so that this PR can merged in? Only two tests are failing currently, which are the floating viewport ones and the search and replace using byte data

Reply:
Hmm let me look at the `Session.search` method, maybe that shouldn't take a `String` and we can (correctly) avoid this encoding issue.